html/pfappserver/root/static.alt/src/views/Nodes/_components/TheForm.vue

```diff
@@ -0,0 +1,801 @@
+</template>
+
+<script>
+import { computed, onBeforeUnmount, onMounted, ref, toRefs, watch } from '@vue/composition-api'
+import { DataSet, Timeline } from 'vue2vis'
+import {
+  BaseForm,
+  BaseFormGroupInput,
+  BaseFormTab
+} from '@/components/new/'
+import pfEmptyTable from '@/components/pfEmptyTable'
+import pfFingerbankScore from '@/components/pfFingerbankScore'
+import pfFormDatetime from '@/components/pfFormDatetime'
+import pfFormRow from '@/components/pfFormRow'
+import { mysqlLimits as sqlLimits } from '@/globals/mysqlLimits'
+import { pfEapType as eapType } from '@/globals/pfEapType'
+import {
+  pfSearchConditionType as conditionType,
+  pfSearchConditionValues as conditionValues
+} from '@/globals/pfSearch'
+import acl from '@/utils/acl'
+import i18n from '@/utils/locale'
+
+import {
+  FormGroupPid,
+  FormGroupStatus,
+  FormGroupRole,
+  FormGroupUnregdate,
+  FormGroupTimeBalance,
+  FormGroupBandwidthBalance,
+  FormGroupVoip,
+  FormGroupBypassVlan,
+  FormGroupBypassRole,
+  FormGroupNotes,
+} from './'
+
+import {
+//   updateValidators,
+  ipLogFields,
+  locationLogFields,
+  securityEventFields,
+  dhcpOption82Fields
+} from '../_config/'
+
+const components = {
+  BaseForm,
+  BaseFormGroupInput,
+  BaseFormTab,
+
+  FormGroupPid,
+  FormGroupStatus,
+  FormGroupRole,
+  FormGroupUnregdate,
+  FormGroupTimeBalance,
+  FormGroupBandwidthBalance,
+  FormGroupVoip,
+  FormGroupBypassVlan,
+  FormGroupBypassRole,
+  FormGroupNotes,
+
+  Timeline,
+  pfEmptyTable,
+  pfFingerbankScore,
+  pfFormDatetime,
+  pfFormRow
+}
+
+const props = {
+  id: { // from router
+    type: String,
+    default: null
+  },
+  form: {
+    type: Object
+  },
+  isLoading: {
+    type: Boolean,
+    default: false
+  }
+}
+
+const setup = (props, context) => {
+
+  const { id, form } = toRefs(props)
+  const { root: { $store } = {} } = context
+
+  if (acl.$can('read', 'security_events')) {
+    $store.dispatch('config/getSecurityEvents')
+  }
+  $store.dispatch('session/getAllowedNodeRoles')
+
+  const iplogSortBy = ref('end_time')
+  const iplogSortDesc =  ref(false)
+
+  const locationSortBy = ref('start_time')
+  const locationSortDesc = ref(true)
+
+  const securityEventSortBy = ref('start_date')
+  const securityEventSortDesc = ref(true)
+
+  const dhcpOption82SortBy = ref('created_at')
+  const dhcpOption82SortDesc = ref(true)
+
+  const triggerSecurityEvent = ref(null)
+  const timelineRef = ref(null)
+
+  const visGroups = new DataSet()
+  const visItems = new DataSet()
+  const visOptions = ref({
+    editable: false,
+    margin: {
+      item: 25
+    },
+    orientation: {
+      axis: 'both',
+      item: 'bottom'
+    },
+    selectable: false,
+    stack: true,
+    tooltip: {
+      followMouse: true
+    }
+  })
+
+  const node = computed(() => $store.state.$_nodes.nodes[id.value])
+  const rolesWithNull = computed(() => {
+    return [
+      { value: null, text: i18n.t('No Role') }, // prepend a null value to roles
```

Comment:
This option may need to be removed. We no longer use `allowEmpty: false` with vue-multiselect and now allow all inputs to be cleared and handle any required values through validations instead. Therefore an additional empty value isn't needed in some cases to help clear the input. The empty option can be left if it helps clarify the choice(s).